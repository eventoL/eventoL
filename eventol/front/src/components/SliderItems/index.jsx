import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import Item from '../Item';
import Slider from '../Slider';

export default class SliderItems extends React.PureComponent {
  static propTypes = {
    itemsData: PropTypes.arrayOf(
      PropTypes.shape({
        attendees: PropTypes.number,
        backdrop: PropTypes.string,
        eventSlug: PropTypes.string.isRequired,
        overview: PropTypes.string.isRequired,
        place: PropTypes.string.isRequired,
        tags: PropTypes.arrayOf(
          PropTypes.shape({
            name: PropTypes.string.isRequired,
            slug: PropTypes.string.isRequired,
          })
        ),
        title: PropTypes.string.isRequired,
        url: PropTypes.string,
      })
    ),
    sliderId: PropTypes.string.isRequired,
  };

  static defaultProps = {
    itemsData: [],
  };

  getItem = itemData => {
    const {sliderId} = this.props;
    return (
      <div key={itemData.key}>
        <Item data={itemData} sliderId={sliderId} />
      </div>
    );
  };

  render() {
    const {itemsData} = this.props;
    let items = '';
    if (!_.isEmpty(itemsData)) {
      items = itemsData.map(this.getItem);
    }
    return <Slider>{items}</Slider>;
  }
}
