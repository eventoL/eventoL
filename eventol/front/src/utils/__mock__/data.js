const place1 = {
  address_components: [
    {long_name: '1667', short_name: '1667', types: ['street_number']},
    {long_name: 'Ayacucho', short_name: 'Ayacucho', types: ['route']},
    {
      long_name: 'Centro',
      short_name: 'Centro',
      types: ['neighborhood', 'political'],
    },
    {
      long_name: 'Rosario',
      short_name: 'Rosario',
      types: ['locality', 'political'],
    },
    {
      long_name: 'Departamento Rosario',
      short_name: 'Departamento Rosario',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Santa Fe',
      short_name: 'Santa Fe',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'S2000', short_name: 'S2000', types: ['postal_code']},
    {long_name: 'BYS', short_name: 'BYS', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Ayacucho 1667, S2000BYS Rosario, Santa Fe, Argentina',
  geometry: {
    location: {lat: -32.9586923, lng: -60.629729},
    location_type: 'ROOFTOP',
    viewport: {
      south: -32.9600412802915,
      west: -60.63107798029148,
      north: -32.9573433197085,
      east: -60.628380019708516,
    },
  },
  place_id: 'ChIJ7QvrMwCrt5URpyBml6doLnY',
  plus_code: {
    compound_code: '29RC+G4 Rosario, Santa Fe, Argentina',
    global_code: '47VX29RC+G4',
  },
  types: ['street_address'],
};

const place2 = {
  address_components: [
    {long_name: '1551', short_name: '1551', types: ['street_number']},
    {long_name: 'Sarmiento', short_name: 'Sarmiento', types: ['route']},
    {
      long_name: 'San Nicolás',
      short_name: 'San Nicolás',
      types: ['political', 'sublocality', 'sublocality_level_1'],
    },
    {
      long_name: 'Comuna 1',
      short_name: 'Comuna 1',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Buenos Aires',
      short_name: 'CABA',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'C1042', short_name: 'C1042', types: ['postal_code']},
    {long_name: 'ABC', short_name: 'ABC', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Sarmiento 1551, C1042ABC CABA, Argentina',
  geometry: {
    location: {lat: -34.6052436, lng: -58.388569700000005},
    location_type: 'ROOFTOP',
    viewport: {
      south: -34.6065925802915,
      west: -58.38991868029149,
      north: -34.60389461970851,
      east: -58.387220719708466,
    },
  },
  place_id: 'ChIJbXpzCcTKvJURbeIVAyh6SnU',
  plus_code: {
    compound_code: '9JV6+WH Buenos Aires, Argentina',
    global_code: '48Q39JV6+WH',
  },
  types: ['street_address'],
};

export const event1 = {
  attendees: 10,
  backdrop: 'event1backdrop',
  eventSlug: 'event1slug',
  overview: 'event1overview',
  place: JSON.stringify(place1),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 1',
  url: 'event1Url',
  id: 1,
};

export const event2 = {
  attendees: 25,
  backdrop: 'event2backdrop',
  eventSlug: 'event2slug',
  overview: 'event2overview',
  place: JSON.stringify(place2),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 2',
  url: 'event2Url',
  id: 2,
};

export const events = [event1, event2];

export default {
  event1,
  event2,
  events,
};
