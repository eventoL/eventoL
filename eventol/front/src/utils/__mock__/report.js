const getConfirmationReport = (confirmed, notConfirmed) => ({
  confirmed,
  not_confirmed: notConfirmed,
  total: confirmed + notConfirmed,
});

export const usersReports = {
  attendee: {
    with_event_user: getConfirmationReport(1, 2),
    without_event_user: getConfirmationReport(2, 3),
  },
  collaborator: getConfirmationReport(3, 4),
  event_user: getConfirmationReport(4, 5),
  installer: getConfirmationReport(6, 7),
  organizer: getConfirmationReport(7, 8),
  speakers: 9,
};

export const activitiesReport = {
  level_count: {
    '1': 3,
    '2': 2,
    '3': 1,
  },
  status_count: {
    '1': 2,
    '2': 1,
  },
  type_count: {
    Talk: 4,
    Workshop: 2,
  },
  ...getConfirmationReport(1, 2),
};

export const installationsReport = {
  hardware_count: {
    notebook: 1,
    phone: 2,
  },
  software_count: {
    ubuntu: 3,
    fedora: 2,
    arch: 1,
  },
  installer_count: {
    '1': 2,
    '2': 1,
  },
  total: 2,
};

export const eventReport = {
  // users data
  ...usersReports,

  // activities data
  activity: activitiesReport,

  // installations data
  installation: installationsReport,
};

export const event1PrivateData = {
  id: 1,
  email: 'organizer1@mailinator.com',
  organizers: 'FirstName1 LastName1 Username1',
};

export const event2PrivateData = {
  id: 2,
  email: 'organizer2@mailinator.com',
  organizers: 'FirstName2 LastName2 Username2',
};

export const eventsPrivateData = [
  {...event1PrivateData},
  {...event2PrivateData},
];

export const getEventData = (id, name, tags, report, eventSlug, location) => ({
  id,
  name,
  tags,
  report,
  location,
  event_slug: eventSlug,
});

export const event1Data = getEventData(
  1,
  'event name',
  {name: 'tag name', slug: 'tag slug'},
  eventReport,
  'event_slug',
  ['Capital Federal', 'Pompeya', 'Buenos Aires', 'Argentina']
);

export const event2Data = getEventData(
  2,
  'event 2 name',
  {name: 'tag 2 name', slug: 'tag 2 slug'},
  eventReport,
  'event_2_slug',
  ['Capital Federal', 'Monserrat', 'Buenos Aires', 'Argentina']
);

export const eventsData = [{...event1Data}, {...event2Data}];

export const expectedActivityDetail = {
  levels: {'1': 6, '2': 4, '3': 2},
  status: {'1': 4, '2': 2},
  types: {Talk: 8, Workshop: 4},
};

export const expectedInstallationsSoftwares = {arch: 2, fedora: 4, ubuntu: 6};

export const expectedTotals = {
  activities: {
    confirmed: 2,
    details: expectedActivityDetail,
    not_confirmed: 4,
    total: 6,
  },
  attendees: {confirmed: 6, not_confirmed: 10, total: 16},
  collaborators: {confirmed: 6, not_confirmed: 8, total: 14},
  installations: {softwares: expectedInstallationsSoftwares, total: 4},
  installers: {confirmed: 12, not_confirmed: 14, total: 26},
  organizers: {confirmed: 14, not_confirmed: 16, total: 30},
  speakers: 18,
};

export const expectedEventDataWithoutPrivateData = {
  ...event1Data,
  assistanceDetail: {attendees: {confirmed: 3, not_confirmed: 5, total: 8}},
  locationDetail: {
    address: 'Pompeya',
    address_detail: 'Capital Federal',
    province: 'Buenos Aires',
  },
};

export const expectedEventDataWithPrivateData = {
  ...expectedEventDataWithoutPrivateData,
  ...event1PrivateData,
};

export const expectedValues = {
  expectedTotals,
  expectedActivityDetail,
  expectedInstallationsSoftwares,
  expectedEventDataWithPrivateData,
  expectedEventDataWithoutPrivateData,
};

const mockedData = {
  activitiesReport,
  eventData: event1Data,
  event1Data,
  event2Data,
  eventsData,
  eventReport,
  event1PrivateData,
  event2PrivateData,
  eventsPrivateData,
  getEventData,
  installationsReport,
  usersReports,
  expectedValues,
};
export default mockedData;
