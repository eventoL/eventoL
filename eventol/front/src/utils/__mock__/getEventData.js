import {eventReport} from './report';

export const getEventData = (id, name, tags, report, eventSlug, location) => ({
  id,
  name,
  tags,
  report,
  location,
  event_slug: eventSlug,
});
export const event1Data = getEventData(
  1,
  'event name',
  {name: 'tag name', slug: 'tag slug'},
  eventReport,
  'event_slug',
  ['Capital Federal', 'Pompeya', 'Buenos Aires', 'Argentina']
);
export const event2Data = getEventData(
  2,
  'event 2 name',
  {name: 'tag 2 name', slug: 'tag 2 slug'},
  eventReport,
  'event_2_slug',
  ['Capital Federal', 'Monserrat', 'Buenos Aires', 'Argentina']
);
export const eventsData = [{...event1Data}, {...event2Data}];
