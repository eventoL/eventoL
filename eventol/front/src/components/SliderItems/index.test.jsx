import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../Item', () => 'Item');
jest.mock('../Slider', () => 'Slider');

import SliderItems from '.';

import {events} from '../../utils/__mock__/data';

describe('SliderItems', () => {
  let component;
  let itemsData;
  let tree;

  const getComponent = () => {
    component = renderer.create(
      <SliderItems itemsData={itemsData} sliderId="id" />
    );
    return component;
  };

  beforeEach(() => {
    itemsData = undefined;
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('With items', () => {
    test('Snapshot', () => {
      itemsData = events;
      component = getComponent();
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
