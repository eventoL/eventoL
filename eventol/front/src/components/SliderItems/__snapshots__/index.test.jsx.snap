// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItems Default Snapshot 1`] = `
<Slider>
  
</Slider>
`;

exports[`SliderItems With items Snapshot 1`] = `
<Slider>
  <div>
    <Item
      data={
        Object {
          "attendees": 10,
          "backdrop": "event1backdrop",
          "eventSlug": "event1slug",
          "id": 1,
          "overview": "event1overview",
          "place": "{\\"address_components\\":[{\\"long_name\\":\\"1667\\",\\"short_name\\":\\"1667\\",\\"types\\":[\\"street_number\\"]},{\\"long_name\\":\\"Ayacucho\\",\\"short_name\\":\\"Ayacucho\\",\\"types\\":[\\"route\\"]},{\\"long_name\\":\\"Centro\\",\\"short_name\\":\\"Centro\\",\\"types\\":[\\"neighborhood\\",\\"political\\"]},{\\"long_name\\":\\"Rosario\\",\\"short_name\\":\\"Rosario\\",\\"types\\":[\\"locality\\",\\"political\\"]},{\\"long_name\\":\\"Departamento Rosario\\",\\"short_name\\":\\"Departamento Rosario\\",\\"types\\":[\\"administrative_area_level_2\\",\\"political\\"]},{\\"long_name\\":\\"Santa Fe\\",\\"short_name\\":\\"Santa Fe\\",\\"types\\":[\\"administrative_area_level_1\\",\\"political\\"]},{\\"long_name\\":\\"Argentina\\",\\"short_name\\":\\"AR\\",\\"types\\":[\\"country\\",\\"political\\"]},{\\"long_name\\":\\"S2000\\",\\"short_name\\":\\"S2000\\",\\"types\\":[\\"postal_code\\"]},{\\"long_name\\":\\"BYS\\",\\"short_name\\":\\"BYS\\",\\"types\\":[\\"postal_code_suffix\\"]}],\\"formatted_address\\":\\"Ayacucho 1667, S2000BYS Rosario, Santa Fe, Argentina\\",\\"geometry\\":{\\"location\\":{\\"lat\\":-32.9586923,\\"lng\\":-60.629729},\\"location_type\\":\\"ROOFTOP\\",\\"viewport\\":{\\"south\\":-32.9600412802915,\\"west\\":-60.63107798029148,\\"north\\":-32.9573433197085,\\"east\\":-60.628380019708516}},\\"place_id\\":\\"ChIJ7QvrMwCrt5URpyBml6doLnY\\",\\"plus_code\\":{\\"compound_code\\":\\"29RC+G4 Rosario, Santa Fe, Argentina\\",\\"global_code\\":\\"47VX29RC+G4\\"},\\"types\\":[\\"street_address\\"]}",
          "tags": Array [
            Object {
              "name": "slug-name",
              "slug": "slug-slug",
            },
          ],
          "title": "event 1",
          "url": "event1Url",
        }
      }
      sliderId="id"
    />
  </div>
  <div>
    <Item
      data={
        Object {
          "attendees": 25,
          "backdrop": "event2backdrop",
          "eventSlug": "event2slug",
          "id": 2,
          "overview": "event2overview",
          "place": "{\\"address_components\\":[{\\"long_name\\":\\"1551\\",\\"short_name\\":\\"1551\\",\\"types\\":[\\"street_number\\"]},{\\"long_name\\":\\"Sarmiento\\",\\"short_name\\":\\"Sarmiento\\",\\"types\\":[\\"route\\"]},{\\"long_name\\":\\"San Nicolás\\",\\"short_name\\":\\"San Nicolás\\",\\"types\\":[\\"political\\",\\"sublocality\\",\\"sublocality_level_1\\"]},{\\"long_name\\":\\"Comuna 1\\",\\"short_name\\":\\"Comuna 1\\",\\"types\\":[\\"administrative_area_level_2\\",\\"political\\"]},{\\"long_name\\":\\"Buenos Aires\\",\\"short_name\\":\\"CABA\\",\\"types\\":[\\"administrative_area_level_1\\",\\"political\\"]},{\\"long_name\\":\\"Argentina\\",\\"short_name\\":\\"AR\\",\\"types\\":[\\"country\\",\\"political\\"]},{\\"long_name\\":\\"C1042\\",\\"short_name\\":\\"C1042\\",\\"types\\":[\\"postal_code\\"]},{\\"long_name\\":\\"ABC\\",\\"short_name\\":\\"ABC\\",\\"types\\":[\\"postal_code_suffix\\"]}],\\"formatted_address\\":\\"Sarmiento 1551, C1042ABC CABA, Argentina\\",\\"geometry\\":{\\"location\\":{\\"lat\\":-34.6052436,\\"lng\\":-58.388569700000005},\\"location_type\\":\\"ROOFTOP\\",\\"viewport\\":{\\"south\\":-34.6065925802915,\\"west\\":-58.38991868029149,\\"north\\":-34.60389461970851,\\"east\\":-58.387220719708466}},\\"place_id\\":\\"ChIJbXpzCcTKvJURbeIVAyh6SnU\\",\\"plus_code\\":{\\"compound_code\\":\\"9JV6+WH Buenos Aires, Argentina\\",\\"global_code\\":\\"48Q39JV6+WH\\"},\\"types\\":[\\"street_address\\"]}",
          "tags": Array [
            Object {
              "name": "slug-name",
              "slug": "slug-slug",
            },
          ],
          "title": "event 2",
          "url": "event2Url",
        }
      }
      sliderId="id"
    />
  </div>
</Slider>
`;
